app/gui/src/main/java/org/mobilitydata/gtfsvalidator/app/gui/GtfsValidatorApp.java

```diff
@@ -0,0 +1,314 @@
+    parent.add(chooseGtfsInputButton);
+    chooseGtfsInputButton.addActionListener(
+        (e) -> {
+          showGtfsInputFileChooser();
+        });
+    parent.add(new JLabel("You can select a ZIP file, a directory, or a URL."));
+  }
+
+  private void showGtfsInputFileChooser() {
+    JFileChooser chooser = new JFileChooser();
+    chooser.setFileSelectionMode(JFileChooser.FILES_AND_DIRECTORIES);
+    chooser.setFileFilter(new GtfsZipsAndDirectoriesFileFilter());
+    int rc = chooser.showOpenDialog(this);
+    if (rc == JFileChooser.APPROVE_OPTION) {
+      File selectedFile = chooser.getSelectedFile();
+      gtfsInputField.setText(selectedFile.toString());
+    }
+  }
+
+  private static class GtfsZipsAndDirectoriesFileFilter extends FileFilter {
+    @Override
+    public boolean accept(File f) {
+      return f.isDirectory() || f.getName().endsWith(".zip");
+    }
+
+    @Override
+    public String getDescription() {
+      return "GTFS ZIPs and Directories";
+    }
+  }
+
+  private void constructOutputDirectorySection(JPanel parent) {
+    // Output Directory Section
+    parent.add(createLabelWithFont("Output Directory:", BOLD_FONT));
+
+    outputDirectoryField
+        .getDocument()
+        .addDocumentListener(documentChangeListener(this::updateValidationButtonStatus));
+    parent.add(outputDirectoryField);
+
+    JButton chooseOutputDirectoryButton = new JButton("Choose Output Directory...");
+    chooseOutputDirectoryButton.addActionListener(
+        (e) -> {
+          showOutputDirectoryChooser();
+        });
+    parent.add(chooseOutputDirectoryButton);
+    parent.add(new JLabel("The validation report will be written here."));
+  }
+
+  private void showOutputDirectoryChooser() {
+    JFileChooser chooser = new JFileChooser();
+    chooser.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+    int rc = chooser.showOpenDialog(this);
+    if (rc == JFileChooser.APPROVE_OPTION) {
+      File selectedFile = chooser.getSelectedFile();
+      outputDirectoryField.setText(selectedFile.toString());
+    }
+  }
+
+  private void constructAdvancedOptionsPanel(JPanel parent) {
+    advancedOptionsPanel.setBorder(
+        BorderFactory.createCompoundBorder(
+            BorderFactory.createEmptyBorder(0, 0, 20, 0),
+            BorderFactory.createTitledBorder("Advanced Options")));
+    parent.add(advancedOptionsPanel);
+
+    JPanel panel = new JPanel();
+    panel.setLayout(new GridBagLayout());
+    advancedOptionsPanel.add(panel);
+
+    GridBagConstraints labelConstraints = new GridBagConstraints();
+    labelConstraints.gridx = 0;
+    labelConstraints.anchor = GridBagConstraints.NORTHWEST;
+    labelConstraints.ipadx = 10;
+
+    GridBagConstraints fieldConstraints = new GridBagConstraints();
+    fieldConstraints.gridx = 1;
+    fieldConstraints.anchor = GridBagConstraints.NORTHEAST;
+
+    labelConstraints.gridy = 0;
+    panel.add(new JLabel("Number of threads:"), labelConstraints);
```

Comment:
The text we chose for the HTML output is "Number of threads used to run the validator", could we match it here, or would it be too long?

Reply:
Done.